### ISSUE TITLE
Crash - zeek-4.1.1-rc1/auxil/broker/caf/libcaf_openssl/src/openssl/session.cpp:133

### ISSUE THREAD

**Issue by initconf** [zeek/zeek#1712]
I noticed that a worker has crashed with a coredump - I think this is new too. 

I have the coredump and related files - if you want them. 

Coredump Anchor: `/usr/local/zeek/spool/tmp/crash-worker-1-1-1627941961-1712`

```
Core was generated by `/usr/local/zeek-4.1.1-rc1/bin/zeek -i enp6s0 -U .status -p zeekctl -p zeekctl-l'.
Program terminated with signal SIGABRT, Aborted.
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:51
51      ../sysdeps/unix/sysv/linux/raise.c: No such file or directory.
[Current thread is 1 (Thread 0x7fb7c565f700 (LWP 10476))]
(gdb) bt
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:51
#1  0x00007fb7d1aef921 in __GI_abort () at abort.c:79
#2  0x00007fb7d4ff672e in ?? () from /usr/lib/x86_64-linux-gnu/libasan.so.4
#3  0x00007fb7d4ffe518 in ?? () from /usr/lib/x86_64-linux-gnu/libasan.so.4
#4  0x00007fb7d4fe9434 in ?? () from /usr/lib/x86_64-linux-gnu/libasan.so.4
#5  0x00007fb7d4ff5896 in ?? () from /usr/lib/x86_64-linux-gnu/libasan.so.4
#6  0x00007fb7d4fdb516 in ?? () from /usr/lib/x86_64-linux-gnu/libasan.so.4
#7  0x00007fb7d4ff7b85 in ?? () from /usr/lib/x86_64-linux-gnu/libasan.so.4
#8  0x00007fb7d4ff7dd1 in ?? () from /usr/lib/x86_64-linux-gnu/libasan.so.4
#9  0x00007fb7d4fe9461 in ?? () from /usr/lib/x86_64-linux-gnu/libasan.so.4
#10 0x00007fb7d4ff5bca in ?? () from /usr/lib/x86_64-linux-gnu/libasan.so.4
#11 0x00007fb7d4f21648 in ?? () from /usr/lib/x86_64-linux-gnu/libasan.so.4
#12 0x00007fb7d4f21a8c in ?? () from /usr/lib/x86_64-linux-gnu/libasan.so.4
#13 0x00007fb7d4f21b5e in ?? () from /usr/lib/x86_64-linux-gnu/libasan.so.4
#14 0x00007fb7d4f1ea4c in ?? () from /usr/lib/x86_64-linux-gnu/libasan.so.4
#15 0x00007fb7d4fd4b0b in malloc () from /usr/lib/x86_64-linux-gnu/libasan.so.4
#16 0x00007fb7d3923f09 in CRYPTO_zalloc () from /usr/lib/x86_64-linux-gnu/libcrypto.so.1.1
#17 0x00007fb7d391f507 in HMAC_CTX_new () from /usr/lib/x86_64-linux-gnu/libcrypto.so.1.1
#18 0x00007fb7d391ee73 in ?? () from /usr/lib/x86_64-linux-gnu/libcrypto.so.1.1
#19 0x00007fb7d391eeff in ?? () from /usr/lib/x86_64-linux-gnu/libcrypto.so.1.1
#20 0x00007fb7d391d765 in EVP_PKEY_CTX_dup () from /usr/lib/x86_64-linux-gnu/libcrypto.so.1.1
#21 0x00007fb7d3905f59 in EVP_MD_CTX_copy_ex () from /usr/lib/x86_64-linux-gnu/libcrypto.so.1.1
#22 0x00007fb7d3c99b8d in ?? () from /usr/lib/x86_64-linux-gnu/libssl.so.1.1
#23 0x00007fb7d3c9b631 in ?? () from /usr/lib/x86_64-linux-gnu/libssl.so.1.1
#24 0x00007fb7d3c986c2 in ?? () from /usr/lib/x86_64-linux-gnu/libssl.so.1.1
#25 0x00007fb7d3c9f995 in ?? () from /usr/lib/x86_64-linux-gnu/libssl.so.1.1
#26 0x00007fb7d3caa252 in ?? () from /usr/lib/x86_64-linux-gnu/libssl.so.1.1
#27 0x00007fb7d3caa373 in SSL_read () from /usr/lib/x86_64-linux-gnu/libssl.so.1.1
#28 0x00007fb7d1698a2a in caf::openssl::session::do_some (this=0x6060002483c0, f=<optimized out>, result=@0x7fb7c565ae30: 0, buf=<optimized out>,
    len=<optimized out>, debug_name=debug_name@entry=0x7fb7d16a00a0 "read_some")
    at /usr/local/src/zeek-4.1.1-rc1/auxil/broker/caf/libcaf_openssl/src/openssl/session.cpp:133
#29 0x00007fb7d16990cc in caf::openssl::session::read_some (this=<optimized out>, result=@0x7fb7c565ae30: 0, buf=<optimized out>, len=<optimized out>)
    at /usr/local/src/zeek-4.1.1-rc1/auxil/broker/caf/libcaf_openssl/src/openssl/session.cpp:145
#30 0x00007fb7d1674b04 in caf::openssl::(anonymous namespace)::ssl_policy::read_some (len=<optimized out>, buf=<optimized out>, fd=<optimized out>,
    result=@0x7fb7c565ae30: 0, this=0x6120003a8460) at /usr/local/src/zeek-4.1.1-rc1/auxil/broker/caf/libcaf_openssl/src/openssl/middleman_actor.cpp:77
#31 caf::io::network::stream::handle_event_impl<caf::openssl::(anonymous namespace)::ssl_policy> (policy=..., op=<optimized out>, this=0x6120003a83b8)
    at /usr/local/src/zeek-4.1.1-rc1/auxil/broker/caf/libcaf_io/caf/io/network/stream.hpp:89
#32 caf::io::network::stream_impl<caf::openssl::(anonymous namespace)::ssl_policy>::handle_event (this=0x6120003a83b8, op=<optimized out>)
    at /usr/local/src/zeek-4.1.1-rc1/auxil/broker/caf/libcaf_io/caf/io/network/stream_impl.hpp:23
#33 0x00007fb7d13845c0 in caf::io::network::default_multiplexer::handle_socket_event (this=this@entry=0x613000004de0, fd=<optimized out>, mask=5,
    ptr=0x6120003a83b8) at /usr/local/src/zeek-4.1.1-rc1/auxil/broker/caf/libcaf_io/src/io/network/default_multiplexer.cpp:554
#34 0x00007fb7d1384b23 in caf::io::network::default_multiplexer::poll_once_impl (this=this@entry=0x613000004de0, block=block@entry=true)
    at /usr/local/src/zeek-4.1.1-rc1/auxil/broker/caf/libcaf_io/src/io/network/default_multiplexer.cpp:191
#35 0x00007fb7d138b955 in caf::io::network::default_multiplexer::poll_once (this=this@entry=0x613000004de0, block=block@entry=true)
    at /usr/local/src/zeek-4.1.1-rc1/auxil/broker/caf/libcaf_io/src/io/network/default_multiplexer.cpp:600
#36 0x00007fb7d138bac8 in caf::io::network::default_multiplexer::run (this=0x613000004de0)
    at /usr/local/src/zeek-4.1.1-rc1/auxil/broker/caf/libcaf_io/src/io/network/default_multiplexer.cpp:201
#37 0x00007fb7d129d3c6 in caf::io::middleman::<lambda()>::operator() (__closure=0x60400124bf30)
    at /usr/local/src/zeek-4.1.1-rc1/auxil/broker/caf/libcaf_io/src/io/middleman.cpp:447
#38 caf::actor_system::<lambda(auto:12)>::operator()<caf::intrusive_ptr<caf::ref_counted> > (guard=..., __closure=0x60400124bf20)
    at /usr/local/src/zeek-4.1.1-rc1/auxil/broker/caf/libcaf_core/caf/actor_system.hpp:555
#39 std::__invoke_impl<void, caf::actor_system::launch_thread(char const*, F) [with F = caf::io::middleman::start()::<lambda()>]::<lambda(auto:12)>, caf::intrusive_ptr<caf::ref_counted> > (__f=...) at /usr/include/c++/7/bits/invoke.h:60
#40 std::__invoke<caf::actor_system::launch_thread(char const*, F) [with F = caf::io::middleman::start()::<lambda()>]::<lambda(auto:12)>, caf::intrusive_ptr<caf::ref_counted> > (__fn=...) at /usr/include/c++/7/bits/invoke.h:95
#41 std::thread::_Invoker<std::tuple<caf::actor_system::launch_thread(char const*, F) [with F = caf::io::middleman::start()::<lambda()>]::<lambda(auto:12)>, caf::intrusive_ptr<caf::ref_counted> > >::_M_invoke<0, 1> (this=0x60400124bf18) at /usr/include/c++/7/thread:234
#42 std::thread::_Invoker<std::tuple<caf::actor_system::launch_thread(char const*, F) [with F = caf::io::middleman::start()::<lambda()>]::<lambda(auto:12)>, caf::intrusive_ptr<caf::ref_counted> > >::operator() (this=0x60400124bf18) at /usr/include/c++/7/thread:243
#43 std::thread::_State_impl<std::thread::_Invoker<std::tuple<caf::actor_system::launch_thread(char const*, F) [with F = caf::io::middleman::start()::<lambda()>]::<lambda(auto:12)>, caf::intrusive_ptr<caf::ref_counted> > > >::_M_run(void) (this=0x60400124bf10) at /usr/include/c++/7/thread:186
#44 0x00007fb7d25136df in ?? () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#45 0x00007fb7d27e66db in start_thread (arg=0x7fb7c565f700) at pthread_create.c:463
#46 0x00007fb7d1bd071f in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

```

**Comment by 0xxon** [comment]
@Neverlord - I assume this might be one for you :)

**Comment by Neverlord** [comment]
@initconf was this the build with the stale Broker submodule reference? Then it might just be the bug in the OpenSSL module that has been fixed with CAF 0.18.5 recently.

**Comment by timwoj** [comment]
We haven't seen this again in further testing. It was likely caused by a version mismatch in the build.
